org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/IVariableProvider.java

```diff
@@ -117,4 +117,19 @@
      * @param priority  the priority for this formatter
      */
     void registerValueFormatter(IValueFormatter formatter, int priority);
+
+    /**
+     * Get the matched value formatter for the specified type.
+     * @param type the type of the valuer
+     * @param options the options about the formatters
+     * @return the matched value formatter, if no formatter matches,
+     * <code>IllegalArgumentException</code> will be thrown
+     */
+    IValueFormatter getValueFormatter(Type type, Map<String, Object> options);
```

Comment:
Do we really need this?
I think the provider already formats the value/type for the caller instead of letting the caller get the formatter.